lib/rspec/support/encoded_string.rb

```diff
@@ -33,21 +45,52 @@ def to_s
 
         private
 
+        ENCODING_STRATEGY = {
+          :bad_bytes => {
+            :invalid => :replace,
+            # :undef   => :nil,
+            :replace => REPLACE
+          },
+          :cannot_convert => {
+            # :invalid => :nil,
+            :undef   => :replace,
+            :replace => REPLACE
+          },
+          :no_converter => {
+            :invalid => :replace,
+            # :undef   => :nil,
+            :replace => REPLACE
+          }
+        }
```

Comment:
Also, is there any real benefit to sticking all these strategies in a big named hash?  Why not just make them separate constants and refer to them separately?  That would remove an unnecessary method call (`Hash#[]`) and it seems like you're just using the hash as an alternate form of named constants.